cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java

```diff
@@ -124,7 +124,7 @@ public void execute(SensorContext context) {
         .append(e)
         .append("'")
         .toString();
-      LOG.error(msg);
+      LOG.error(msg, e);

```

Comment:
No need for this. There is the `.append(e)` why again in `.error`?

Reply:
add also a check to  `((NSString) filePath).getContent()`

Reply:
Catch requireNonNull exception

Reply:
I commented about this earlier [here](https://github.com/SonarOpenCommunity/sonar-cxx/issues/1442#issuecomment-378314000). In situation when any _[unexpected](https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-sensors/src/main/java/org/sonar/cxx/sensors/clangsa/CxxClangSASensor.java#L109)_ exception happens stack trace is never printed. Because `.append(e)` only appends error message without stack trace. Same happens here in `LOG.error(msg)` which again only prints error message without stack trace. Note that `LOG.error(msg)` was there before me. Increasing verbosity and enabling debug does not help here, I had to modify this line in my custom plugin to see what is going on.